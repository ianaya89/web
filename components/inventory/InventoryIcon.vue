<template>
  <svg-icon class="inline" :name="'inventory/'+getMarketplaceIcon" />
</template>

<script>
import { getMarketplaceCategoryIcon } from '@/services/helpers'

export default {
  name: 'InventoryIcon',

  props: {
    category: {
      type: Object,
      required: true
    },
    isPackage: {
      type: Boolean,
      default: false
    }
  },

  computed: {
    getMarketplaceIcon() {
      if (
        this.category == null ||
        !Object.prototype.hasOwnProperty.call(this.category, 'path')
      ) {
        return 'inventory'
      }

      if (this.isPackage) {
        return getMarketplaceCategoryIcon('package / ' + this.category.path)
      } else {
        return getMarketplaceCategoryIcon(this.category.path)
      }
    }
  }
}
</script>

<style scoped>
</style>
