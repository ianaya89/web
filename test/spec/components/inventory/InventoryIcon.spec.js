import { shallowMount } from '@vue/test-utils'
import InventoryIcon from '@/components/inventory/InventoryIcon'

describe('InventoryIcon', () => {
  it('renders default inventory icon', () => {
    const wrapper = shallowMount(InventoryIcon)
    expect(wrapper.element).toMatchSnapshot()
  })

  it('renders package inventory icon', () => {
    const wrapper = shallowMount(InventoryIcon, {
      propsData: {
        category: {
          path: ''
        },
        isPackage: true
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
