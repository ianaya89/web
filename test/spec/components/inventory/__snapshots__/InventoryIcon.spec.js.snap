// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InventoryIcon renders default inventory icon 1`] = `
<svg-icon
  class="inline"
  name="inventory/inventory"
/>
`;

exports[`InventoryIcon renders package inventory icon 1`] = `
<svg-icon
  class="inline"
  name="inventory/package"
/>
`;
